tests/testthat/test-coerce.R

```diff
@@ -43,6 +43,10 @@ test_that("can coerce to character vectors", {
   expect_equal(coerce_chr("x"), "x")
 })
 
+test_that("error captures correct env", {
```

Comment:
To really test that the correct env is captured, I think this needs to simulate an indirect usage. Then the snapshot would show specific advice.

I think we should make it easier to test. Something like:

```r
expect_deprecated(foo(), type = c("any", "direct", "indirect"))
```